packages/common/src/transport/alchemy.ts

```diff
@@ -169,112 +126,56 @@ export function alchemy<
     "Alchemy-AA-Sdk-Version": VERSION,
   };
 
-  if (connectionConfig.jwt != null || connectionConfig.apiKey != null) {
+  if ("jwt" in connectionConfig) {
+    fetchOptions.headers = {
+      ...fetchOptions.headers,
+      Authorization: `Bearer ${connectionConfig.jwt}`,
+    };
+  } else if ("apiKey" in connectionConfig) {
     fetchOptions.headers = {
       ...fetchOptions.headers,
-      Authorization: `Bearer ${
-        connectionConfig.jwt ?? connectionConfig.apiKey
-      }`,
+      Authorization: `Bearer ${connectionConfig.apiKey}`,
     };
   }
 
   const transport: AlchemyTransportBase = (opts) => {
     const { chain } = opts;
-    if (!chain) {
-      throw new ChainNotFoundError();
-    }
 
-    const { innerTransport, rpcUrl } = (() => {
-      mutateRemoveTrackingHeaders(config?.fetchOptions?.headers);
-      if (
-        connectionConfig.overrides != null &&
-        connectionConfig.fallback != null
-      ) {
-        return {
-          innerTransport: split(connectionConfig),
-          rpcUrl:
-            connectionConfig.restConnection.proxyUrl ??
-            "https://api.g.alchemy.com/v2",
-        };
-      }
+    mutateRemoveTrackingHeaders(config?.fetchOptions?.headers);
 
-      if (!connectionConfig.proxyUrl && chain.rpcUrls.alchemy == null) {
-        // TODO(v5): update this error message to be the correct package name
-        throw new BaseError(
-          "chain must include an alchemy rpc url. See `defineAlchemyChain` or import a chain from `@alchemy/common/chains`.",
-        );
+    const rpcUrl = (() => {
+      if ("url" in connectionConfig) {
+        return connectionConfig.url;
```

Comment:
when this transport is used in chain-agnostic settings (since the route splitting logic is moving elsewhere), will they always need to specify the exact `url` including the api key?

Reply:
ah yeah that is a good catch thanks. changed to allow apikey or jwt with `url` and attaching it as a header